moved/src/move_execution/evm_native.rs

```diff
@@ -0,0 +1,862 @@
+    let caller = safely_pop_arg!(args, AccountAddress);
+
+    evm_transact_inner(
+        context,
+        caller.to_eth_address(),
+        TxKind::Create,
+        from_move_u256(value),
+        data,
+    )
+}
+
+fn evm_transact_inner(
+    context: &mut SafeNativeContext,
+    caller: Address,
+    transact_to: TxKind,
+    value: U256,
+    data: Vec<u8>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    // TODO: does it make sense for EVM gas to be 1:1 with MoveVM gas?
+    let gas_limit: u64 = context.gas_balance().into();
+
+    let evm_native_ctx = context.extensions_mut().get_mut::<NativeEVMContext>();
+    let mut evm = Evm::builder()
+        .with_db(&mut evm_native_ctx.db)
+        .with_tx_env(TxEnv {
+            caller,
+            gas_limit,
+            // Gas price can be zero here because fee is charged in the MoveVM
+            gas_price: U256::ZERO,
+            transact_to,
+            value,
+            data: data.into(),
+            // Nonce and chain id can be None because replay attacks
+            // are prevented at the MoveVM level. I.e. replay will
+            // never occur because the MoveVM will not accept a duplicate
+            // transaction
+            nonce: None,
+            chain_id: None,
+            // TODO: could maybe construct something based on the values that
+            // have already been accessed in `context.traversal_context()`.
+            access_list: Vec::new(),
+            gas_priority_fee: None,
+            blob_hashes: Vec::new(),
+            max_fee_per_blob_gas: None,
+            authorization_list: None,
+        })
+        .with_block_env(BlockEnv {
+            number: U256::from(evm_native_ctx.block_header.number),
+            coinbase: Address::ZERO,
+            timestamp: U256::from(evm_native_ctx.block_header.timestamp),
+            gas_limit: U256::from(u64::MAX),
+            basefee: U256::ZERO,
+            difficulty: U256::ZERO,
+            prevrandao: Some(evm_native_ctx.block_header.prev_randao),
+            blob_excess_gas_and_price: Some(BlobExcessGasAndPrice {
+                excess_blob_gas: 0,
+                blob_gasprice: 0,
+            }),
+        })
+        .build();
+
+    let outcome = evm.transact().map_err(|e| match e {
+        EVMError::Database(e) => SafeNativeError::InvariantViolation(e),
+        other => SafeNativeError::InvariantViolation(
+            PartialVMError::new(StatusCode::ABORTED).with_message(format!("EVM Error: {other:?}")),
+        ),
+    })?;
+    drop(evm);
+
+    // Capture changes in native context so that they can be
+    // converted into Move changes when the session is finalized
+    evm_native_ctx.state_changes.push(outcome.state.clone());
+
+    // Commit the changes to the DB so that future Move transactions using
+    // the same session will see them.
+    evm_native_ctx.db.commit(outcome.state);
+
+    let gas_used = EvmGasUsed::new(outcome.result.gas_used());
```

Comment:
We'll probably use how much it costed using Move resources than passing the EVM cost over. But I think this is a good way to start to compare them later to decide what makes more sense.